src/checkpatch_wrapper.sh

```diff
@@ -12,54 +20,56 @@ function execute_checkpatch()
 
   local -r options="--terse --no-tree --color=always -strict --file "
   local -r script="scripts/checkpatch.pl $options"
-  local -r original_working_dir=$PWD
+  local -r original_working_dir="$PWD"
   local kernel_root=""
 
-  if [[ -z $FILE_OR_DIR_CHECK ]]; then
+  if [[ -z "$FILE_OR_DIR_CHECK" ]]; then
     FILE_OR_DIR_CHECK="."
   fi
 
-  FILE_OR_DIR_CHECK="$(realpath $FILE_OR_DIR_CHECK)"
   # Check if is a valid path
-  if [[ ! -d $FILE_OR_DIR_CHECK && ! -f $FILE_OR_DIR_CHECK ]]; then
+  if [[ ! -d "$FILE_OR_DIR_CHECK" && ! -f "$FILE_OR_DIR_CHECK" ]]; then
     complain "Invalid path"
-    return 1
+    return 22 # EINVAL
   fi
 
-  # try to find kernel root at given path
+  # Get realpath for using inside checkpatch
+  FILE_OR_DIR_CHECK="$(realpath $FILE_OR_DIR_CHECK)"
+
+  # Try to find kernel root at given path
   kernel_root="$(find_kernel_root $FILE_OR_DIR_CHECK)"
   if [[ -z "$kernel_root" ]]; then
-    # fallback: try to find kernel root at working path
+    # Fallback: try to find kernel root at working path
     kernel_root="$(find_kernel_root $original_working_dir)"
   fi
 
-  # Check if kernel root was found.
+  # Check if kernel root was found
   if [[ -z "$kernel_root" ]]; then
     complain "Neither the given path nor the working path is in a kernel tree."
-    return 1
+    return 22 # EINVAL
   fi
+
   # Build a list of file to apply check patch
-  FLIST=`find $FILE_OR_DIR_CHECK -type f ! -name '*\.mod\.c' | grep "\.[ch]$" `
+  cmd="find $FILE_OR_DIR_CHECK -type f ! -name '*\.mod\.c' | grep \"\.[ch]$\""
+  FLIST=$(cmd_manager "SILENT" "$cmd")
+
   say "Running checkpatch.pl on: $FILE_OR_DIR_CHECK"
-  say $SEPARATOR
+  say "$SEPARATOR"
 
-  for current_file in $FLIST
-  do
-    file=$current_file
+  cmd="perl $script "
+  for current_file in $FLIST; do
+    file="$current_file"
 
-    if [ ! -e "$file" ]
-    then
-      echo "$file does not exist."; echo
+    if [[ ! -e "$file" ]]; then
+      echo -e "$file does not exist. \n"
       continue
     fi
 
-    cd $kernel_root
-    eval perl $script $file
-    cd $original_working_dir
+    cd "$kernel_root"
 
-    if [ $? != 0 ]; then
-      say $SEPARATOR
-    fi
+    cmd_manager "SILENT" "$cmd $file"
+    say "$SEPARATOR"
```

Comment:
In #43 you mentioned no output would be preferable when there's no codestyle errors. I.e. something like:

```
$ kw c drivers/gpu/drm/vkms
$
```

However, with the patch, we may get outputs such as:

```
$ kw c drivers/gpu/drm/vkms
Running checkpatch.pl on: linux/drivers/gpu/drm/vkms
=========================================================
=========================================================
=========================================================
=========================================================
=========================================================
```

For me it's OK like that! I just wanted to comment this to see if it is OK for you too.